tests/cypress/e2e/for-teacher_page/class_page/second_teachers_invites.cy.js

```diff
@@ -1,83 +1,81 @@
+// They cannot be made independent until https://github.com/hedyorg/hedy/issues/4804 is resolved
 describe("Second teachers: invitations", () => {
+  // Before all: create a single class
+  let className;
   before(() => {
     loginForTeacher();
-    createClass()
+    className = createClass()
   });
 
   it(`Invites ${secondTeachers.length} second teachers: by username`, () => {
     loginForTeacher();
-    cy.get(".view_class").first().click();
+    navigateToClass(className);
 
     for (const teacher of secondTeachers) {
-      
       cy.get("#add-second-teacher").click();
       cy.get("#modal-prompt-input").type(teacher);
       cy.get("#modal-ok-button").click();
-      
-      cy.get("body").then(invitesTable).then(table => {
-        if (table.length) {
-          table = cy.get("#invites-block table")
-          cy.get("#invites-block .username_cell")
-          .should('be.visible')
-          .and("include.text", teacher)
-        } else {
-          cy.log("Second teacher not invited.")
-        }
-      })
+    }
+
+    // Check that both invited teachers are in the table
+    for (const teacher of secondTeachers) {
+      cy.get("#invites-block table")
+        .contains(teacher);
     }
   })
 
   it(`Tries duplicating ${secondTeachers[0]}'s invitation`, () => {
     loginForTeacher();
-    cy.get(".view_class").first().click();
+    navigateToClass(className);
 
     cy.get("#add-second-teacher").click();
     cy.get("#modal-prompt-input").type(secondTeachers[0]);
     cy.get("#modal-ok-button").click();
 
     cy.get("#modal_alert_container")
-    .should('be.visible')
-    .and("contain", "pending invitation")
+      .contains('pending invitation')
   })
 
+  it(`Deletes ${secondTeachers[1]}'s invitation`, () => {
+    loginForTeacher();
+    navigateToClass(className);
+
+    cy.get("#invites-block table")
+      .get('.username_cell')
+      .contains(secondTeachers[1])
+      .parent('tr')
+      .find('.remove_user_invitation a')
+      .click();
+
+    cy.get('#modal-confirm #modal-yes-button').click();
+
+    // This needs to come before we accept teacher2's invitation, otherwise
+    // after this there are no invites and so this table isn't rendered at all.
+    cy.get("#invites-block table")
+      .should("not.contain", secondTeachers[1]);
+  });
+
   it(`Accepts invitation sent to ${secondTeachers[0]}`, () => {
-    loginForTeacher(secondTeachers[0])
-    goToProfilePage()
-    cy.get("#messages").should("exist")
-    cy.get("#messages #join").click()
-  })
+    loginForTeacher(secondTeachers[0]);
+    goToProfilePage();
+    cy.get("#messages #join").click();
 
-  it("Reads all second teachers", () => {
-    loginForTeacher()
-    cy.get(".view_class").first().click();        
-    cy.get("#second_teachers_container .username_cell").should("include.text", secondTeachers[0])
+    // Give the Ajax request that gets sent as a result of the click enough time to complete
+    cy.wait(500);
```

Comment:
What if instead of adding a hardcoded wait we do

```
cy.intercept('/class/join/*').as('join')
// test code
cy.wait('@join')
```

Reply:
Dunno, never used it. I like it better, but not behind my machine right now.

do you want to make the change?

Reply:
(And if you do, feel free to ship it 🤪🥸)

Reply:
Yes, I'll do the change!